driver-core/src/test/functional/com/mongodb/client/model/OperationTest.java

```diff
@@ -110,6 +115,37 @@ protected void assertResults(final List<Bson> pipeline, final String expectedRes
         assertEquals(expectedResults, results);
     }
 
+    protected void assertResults(final List<Bson> pipeline, final String expectedResultsAsString,
+            final int scale, final RoundingMode roundingMode) {
+        List<BsonDocument> expectedResults = parseToList(expectedResultsAsString);
+        List<BsonDocument> results = getCollectionHelper().aggregate(pipeline);
+        assertEquals(adjustScale(expectedResults, scale, roundingMode), adjustScale(results, scale, roundingMode));
```

Comment:
Looks good. Another option would have been to perform recursive equality, with an epsilon on floating points. But I don't think we'll have an issue in these tests since the decimal places aren't close to being 0 (which would make the flooring flaky).